app/controllers/new_user/dossiers_controller.rb

```diff
@@ -95,16 +97,26 @@ def dossier
     end
 
     def dossier_with_champs
-      @dossier_with_champs ||= current_user.dossiers.with_ordered_champs.find(params[:id])
+      @dossier_with_champs ||= Dossier.with_ordered_champs.find(params[:id])
     end
 
     def ensure_ownership!
       if dossier.user_id != current_user.id
-        flash[:alert] = "Vous n'avez pas accès à ce dossier"
-        redirect_to root_path
+        forbidden!
       end
     end
 
+    def ensure_ownership_or_invitation!
+      if !dossier.owner_or_invite?(current_user)
+        forbidden!
+      end
+    end
+
+    def forbidden!
```

Comment:
Non bloquant : le maniaque en moi aurait extrait l'introduction de cette méthode et son utilisation dans le code actuel dans un commit qui précède `[#1677] Allow invites to edit dossiers ` histoire de rendre la lecture de ce dernier plus fluide